packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts

```diff
@@ -128,66 +199,121 @@ export class ConnectionTestDataState {
+            ),
+          },
+          report,
+        )) as string;
+    } else {
+      // TODO: delete this once the backend code is in place
+      value =
+        (yield this.editorStore.graphManagerState.graphManager.generateExecuteTestData(
+          serviceExecutionParameters.query,
+          buildTestDataParameters(
+            serviceExecutionParameters.query,
+            this.editorStore,
+          ),
+          serviceExecutionParameters.mapping,
+          serviceExecutionParameters.runtime,
+          this.editorStore.graphManagerState.graph,
+          {
+            anonymizeGeneratedData: this.anonymizeGeneratedData,
+          },
+          report,
+        )) as string;
+    }
+    // NOTE: since we don't have a generic mechanism for test-data generation
+    // we will only report metrics around API usage, when we genericize, we will
+    // move this out
+    LegendStudioTelemetryHelper.logEvent_TestDataGenerationSucceeded(
+      this.editorStore.applicationStore.telemetryService,
+      report,
+    );
+    service_setConnectionTestDataEmbeddedData(
+      this.connectionData,
+      TEMPORARY__createRelationalDataFromCSV(value),
+      this.editorStore.changeDetectionState.observerContext,
+    );
+  }
+
   *generateTestData(): GeneratorFn<void> {
     try {
       this.generatingTestDataSate.inProgress();
       const connection = guaranteeNonNullable(
         this.resolveConnectionValue(this.connectionData.connectionId),
         `Unable to resolve connection ID '${this.connectionData.connectionId}`,
       );
-
-      let embeddedData: EmbeddedData;
       if (connection instanceof DatabaseConnection) {
         const serviceExecutionParameters = guaranteeNonNullable(
           this.testDataState.testSuiteState.testableState.serviceEditorState
             .executionState.serviceExecutionParameters,
         );
-
-        // NOTE: since we don't have a generic mechanism for test-data generation
-        // we will only report metrics around API usage, when we genericize, we will
-        // move this out
-        LegendStudioTelemetryHelper.logEvent_TestDataGenerationLaunched(
-          this.testDataState.editorStore.applicationStore.telemetryService,
-        );
-        const report = reportGraphAnalytics(
-          this.editorStore.graphManagerState.graph,
-        );
-
-        const value =
-          (yield this.editorStore.graphManagerState.graphManager.generateExecuteTestData(
-            serviceExecutionParameters.query,
-            buildTestDataParameters(
-              serviceExecutionParameters.query,
-              this.editorStore,
+        if (
+          this.editorStore.applicationStore.config.options
+            .TEMPORARY__enableTestDataGenerationNewFlow
+        ) {
+          const parameters = (serviceExecutionParameters.query.parameters ??
+            []) as object[];
+          if (parameters.length) {
+            this.parameterState.openModal(serviceExecutionParameters);
```

Comment:
dont we want to return after this ? otherwise we continue with the flow below.

Reply:
Thanks. 1. added return 2. added try catch